### ISSUE TITLE
./test_signals fails with fatal error

### ISSUE THREAD

**Issue by ColinGilbert** [cdglove/entity#1]
Hi,

Running Gentoo with Boost 1.56 installed. Had to disable variadics in order to compile. All tests seem to work except ./test_signals. Here is the error message:

```
./test_signals
Running 2 test cases...
unknown location(0): fatal error in "auto_component_creation": memory access violation at address: 0x00000000: no mapping at fault address
/home/noob/CoolCode/entity/test/signals.cpp(37): last checkpoint

*** 1 failure detected in test suite "Signals"
```

Please ask if you would like more info in order to debug/reproduce.

Colin

**Comment by cdglove** [comment]
Hi Colin, 

Can you please let me know what compiler you're using? I find it odd you had to disable variadics as that define is primarily for Visual Studio 2010 support.

-- chris

**Comment by ColinGilbert** [comment]
I believe it was gcc 4.8.3

**Comment by ColinGilbert** [comment]
Here is another go with the updated (as of today) repo. Unfortunately, it fails. :(

```
noob@NoobKing ~/CoolCode/entity/build $ cmake ..
-- The C compiler identification is GNU 4.8.3
-- The CXX compiler identification is GNU 4.8.3
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test COMPILER_SUPPORTS_CXX11
-- Performing Test COMPILER_SUPPORTS_CXX11 - Success
-- Found Git: /usr/bin/git (found version "2.0.5") 
-- Cloning repo https://github.com/cdglove/daily.git
Cloning into '/home/noob/CoolCode/entity/contrib/daily'...
Daily Version: 1.0.0
-- Boost version: 1.56.0
-- Boost version: 1.56.0
-- Found the following Boost libraries:
--   unit_test_framework
--   system
--   thread
--   date_time
--   timer
--   chrono
-- Configuring done
-- Generating done
-- Build files have been written to: /home/noob/CoolCode/entity/build
noob@NoobKing ~/CoolCode/entity/build $ make
Scanning dependencies of target entity
[  5%] Building CXX object CMakeFiles/entity.dir/src/dummy.cpp.o
Linking CXX static library libentity.a
[  5%] Built target entity
Scanning dependencies of target daily
[ 10%] Building CXX object contrib/daily/src/CMakeFiles/daily.dir/fast_iostream/fast_locale.cpp.o
[ 15%] Building CXX object contrib/daily/src/CMakeFiles/daily.dir/trace/log.cpp.o
[ 20%] Building CXX object contrib/daily/src/CMakeFiles/daily.dir/trace/boost_assert.cpp.o
Linking CXX static library libdaily.a
[ 20%] Built target daily
Scanning dependencies of target test_basic_compilation
[ 25%] Building CXX object test/CMakeFiles/test_basic_compilation.dir/instantiate_pools.cpp.o
In file included from /home/noob/CoolCode/entity/test/instantiate_pools.cpp:1:0:
/home/noob/CoolCode/entity/entity/dense_component_pool.hpp: In member function ‘void entity::dense_component_pool<T>::auto_create_components(entity::entity_pool&, Args&& ...)’:
/home/noob/CoolCode/entity/entity/dense_component_pool.hpp:250:33: error: parameter packs not expanded with ‘...’:
       [this, constructor_args...](entity e)
                                 ^
/home/noob/CoolCode/entity/entity/dense_component_pool.hpp:250:33: note:         ‘constructor_args’
/home/noob/CoolCode/entity/entity/dense_component_pool.hpp:249:33: error: expected primary-expression before ‘(’ token
      std::function<void(entity)>(
                                 ^
/home/noob/CoolCode/entity/entity/dense_component_pool.hpp:250:30: error: expected ‘,’ before ‘...’ token
       [this, constructor_args...](entity e)
                              ^
/home/noob/CoolCode/entity/entity/dense_component_pool.hpp:250:30: error: expected identifier before ‘...’ token
/home/noob/CoolCode/entity/entity/dense_component_pool.hpp:250:33: error: parameter packs not expanded with ‘...’:
       [this, constructor_args...](entity e)
                                 ^
/home/noob/CoolCode/entity/entity/dense_component_pool.hpp:250:33: note:         ‘constructor_args’
/home/noob/CoolCode/entity/entity/dense_component_pool.hpp: In lambda function:
/home/noob/CoolCode/entity/entity/dense_component_pool.hpp:253:34: error: expansion pattern ‘constructor_args’ contains no argument packs
        create(e, constructor_args...);
                                  ^
In file included from /home/noob/CoolCode/entity/test/instantiate_pools.cpp:2:0:
/home/noob/CoolCode/entity/entity/sparse_component_pool.hpp: In member function ‘void entity::sparse_component_pool<T>::auto_create_components(entity::entity_pool&, Args&& ...)’:
/home/noob/CoolCode/entity/entity/sparse_component_pool.hpp:211:33: error: parameter packs not expanded with ‘...’:
       [this, constructor_args...](entity e)
                                 ^
/home/noob/CoolCode/entity/entity/sparse_component_pool.hpp:211:33: note:         ‘constructor_args’
/home/noob/CoolCode/entity/entity/sparse_component_pool.hpp:210:33: error: expected primary-expression before ‘(’ token
      std::function<void(entity)>(
                                 ^
/home/noob/CoolCode/entity/entity/sparse_component_pool.hpp:211:30: error: expected ‘,’ before ‘...’ token
       [this, constructor_args...](entity e)
                              ^
/home/noob/CoolCode/entity/entity/sparse_component_pool.hpp:211:30: error: expected identifier before ‘...’ token
/home/noob/CoolCode/entity/entity/sparse_component_pool.hpp:211:33: error: parameter packs not expanded with ‘...’:
       [this, constructor_args...](entity e)
                                 ^
/home/noob/CoolCode/entity/entity/sparse_component_pool.hpp:211:33: note:         ‘constructor_args’
/home/noob/CoolCode/entity/entity/sparse_component_pool.hpp: In lambda function:
/home/noob/CoolCode/entity/entity/sparse_component_pool.hpp:213:34: error: expansion pattern ‘constructor_args’ contains no argument packs
        create(e, constructor_args...);
                                  ^
In file included from /home/noob/CoolCode/entity/test/instantiate_pools.cpp:3:0:
/home/noob/CoolCode/entity/entity/saturated_component_pool.hpp: In member function ‘void entity::saturated_component_pool<T>::auto_create_components(entity::entity_pool&, Args ...)’:
/home/noob/CoolCode/entity/entity/saturated_component_pool.hpp:244:33: error: parameter packs not expanded with ‘...’:
       [this, constructor_args...](entity e) 
                                 ^
/home/noob/CoolCode/entity/entity/saturated_component_pool.hpp:244:33: note:         ‘constructor_args’
/home/noob/CoolCode/entity/entity/saturated_component_pool.hpp:243:33: error: expected primary-expression before ‘(’ token
      std::function<void(entity)>(
                                 ^
/home/noob/CoolCode/entity/entity/saturated_component_pool.hpp:244:30: error: expected ‘,’ before ‘...’ token
       [this, constructor_args...](entity e) 
                              ^
/home/noob/CoolCode/entity/entity/saturated_component_pool.hpp:244:30: error: expected identifier before ‘...’ token
/home/noob/CoolCode/entity/entity/saturated_component_pool.hpp:244:33: error: parameter packs not expanded with ‘...’:
       [this, constructor_args...](entity e) 
                                 ^
/home/noob/CoolCode/entity/entity/saturated_component_pool.hpp:244:33: note:         ‘constructor_args’
/home/noob/CoolCode/entity/entity/saturated_component_pool.hpp: In lambda function:
/home/noob/CoolCode/entity/entity/saturated_component_pool.hpp:246:34: error: expansion pattern ‘constructor_args’ contains no argument packs
        create(e, constructor_args...);
                                  ^
test/CMakeFiles/test_basic_compilation.dir/build.make:54: recipe for target 'test/CMakeFiles/test_basic_compilation.dir/instantiate_pools.cpp.o' failed
make[2]: *** [test/CMakeFiles/test_basic_compilation.dir/instantiate_pools.cpp.o] Error 1
CMakeFiles/Makefile2:205: recipe for target 'test/CMakeFiles/test_basic_compilation.dir/all' failed
make[1]: *** [test/CMakeFiles/test_basic_compilation.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by ColinGilbert** [comment]
Update: Compiles if you disable variadics. Again.

test_signals still fails :(

```
noob@NoobKing ~/CoolCode/entity/build/test $ ./test_signals
Running 2 test cases...
unknown location(0): fatal error in "auto_component_creation": memory access violation at address: 0x00000000: no mapping at fault address
/home/noob/CoolCode/entity/test/signals.cpp(37): last checkpoint

*** 1 failure detected in test suite "Signals"
```

**Comment by cdglove** [comment]
Thanks!  Both are weird, I'll take a look when I get a chance.  Probably not until the weekend though.

**Comment by cdglove** [comment]
Fixed as of commit 38478a524ca930663ce5248b922cc0af47c10a58.  The bug was due to some rotting of the ENTITY_SUPPORT_VARIADICS == 0 code.

Also, found the issue with variadics in gcc-4.8.  Turns out it that version won't lambda capture a parameter pack, so I disabled variadic support for gcc < 4.9 in commit 603faf21efe36bf79370082a4cee2d052d6abf6a.  

I've added a build entry on travis-ci for gcc-4.8 so this doesn't rot in the future.
